### ISSUE TITLE
SQL Translation Error

### ISSUE THREAD

**Issue by cfmitrah** [jOOQ/jOOQ#9845]
### SQL translator incorrectly translates this query:

```sql
SELECT i.*, SUBSTRING(
    (
        SELECT  DISTINCT '-'    UPPER(im.gu_market) AS "text()"
		FROM 
            Z_INCOTERMS_MARKETS IM
		WHERE 
            i.gu_incoterm = im.gu_incoterm
        ORDER BY "text()"
		FOR XML Path('')
    ), 2, 10000
) AS marketlist
FROM   
    z_incoterms i
WHERE  
    i.bo_active = 1
```

### Parser output:

```sql
UPDATE, NO KEY UPDATE, SHARE, or KEY SHARE expected: [9:7] ... = im.gu_incoterm
        ORDER BY "text()"
		FOR [*]XML Path('')
    ), 2, 10000
) AS marketlist
FROM   
    z_incoterms i
WHERE  
 ...
```

**Comment by lukaseder** [comment]
Thank you very much for your report. We currently do not support SQL Server's `FOR XML` and `FOR JSON` syntax yet, but it would be a good idea to do so. I've created a feature request for this:
https://github.com/jOOQ/jOOQ/issues/9923

**Comment by lukaseder** [comment]
Parsing and API support has been implemented in #9923, and will ship soon. Emulations, which will be required for translations will be implemented in #9924. `FOR JSON` emulations have been implemented already: #9926.

I'm closing this as a duplicate of the other issues.
